cmd/vm_firewall.go

```diff
@@ -125,26 +137,25 @@ func setVirtualMachineSecurityGroups(vm *egoscale.VirtualMachine, sgs []egoscale
 		ids[i] = *sgs[i].ID
 	}
 
-	resp, err := cs.RequestWithContext(gContext, &egoscale.UpdateVirtualMachine{
+	_, err := cs.RequestWithContext(gContext, &egoscale.UpdateVirtualMachine{
 		ID:               vm.ID,
 		SecurityGroupIDs: ids,
 	})
 	if err != nil {
 		return err
 	}
 
-	vm, ok := resp.(*egoscale.VirtualMachine)
-	if !ok {
-		return fmt.Errorf("wrong type expected %q, got %T", "egoscale.VirtualMachine", resp)
-	}
-
 	return nil
 }
 
 // printVirtualMachineSecurityGroups prints a virtual machine instance security groups to standard output.
 func printVirtualMachineSecurityGroups(vm *egoscale.VirtualMachine) {
 	// Refresh the vm object to ensure its properties are up-to-date
-	vm, _ = getVirtualMachineByNameOrID(vm.ID.String())
+	vm, err := getVirtualMachineByNameOrID(vm.ID.String())
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "%v\n", err)
+		os.Exit(1)
```

Comment:
:fearful:

Reply:
https://github.com/exoscale/cli/blob/master/cmd/sos_remove.go#L96

Feel free to rewrite this as you see fit.

Reply:
blaming other parts of the codebase won't help.